src/merge.rs

```diff
@@ -86,8 +101,8 @@ enum ConflictResolution {
 /// nested hierarchies, or make conflicting changes on multiple devices
 /// simultaneously. A simpler two-way tree merge strikes a good balance between
 /// correctness and complexity.
-#[derive(Debug)]
 pub struct Merger<'t> {
+    driver: &'t Driver,
```

Comment:
Default params should still work. Possibly worth noting that you use them frequently, HashMap<K, V> is actually HashMap<K, V, S = RandomState>. You'll probably have to add `new` (e.g. the constructor that doesn't take a Driver arg) to `impl<'a> Merger<'a, DefaultDriver>` instead of `impl<'a, D: Driver> Merger<'a, D>`, to avoid the issue you described though.